rust/src/dns/dns.rs

```diff
@@ -821,9 +917,16 @@ pub extern "C" fn rs_dns_tx_get_query_rrtype(tx: &mut DNSTransaction,
 }
 
 #[no_mangle]
-pub extern "C" fn rs_dns_probe(input: *const u8, len: u32, rdir: *mut u8)
-                               -> u8
-{
+pub extern "C" fn rs_dns_probe(
+    _flow: *const core::Flow,
+    _dir: u8,
+    input: *const u8,
+    len: u32,
+    rdir: *mut u8,
+) -> AppProto {
+    if len == 0 || len < std::mem::size_of::<DNSHeader>() as u32 {
+        return core::ALPROTO_UNKNOWN;
+    }
     let slice: &[u8] = unsafe {
```

Comment:
we can use the `build_slice!` macro here.

Reply:
We could, but then I'd get an error about the function being unsafe.  Maybe could just be marked as a TODO til then.

Reply:
ok

Reply:
Oh, but I suppose I should actually make the function unsafe to support my argument ;)